File: IagoAuction/ClientApp/src/app/shared/dto/auction-dto.ts
export class AuctionDto {
  id: number;
  title: string;
  description: string;
  paintingIds: number[]
  startDate: Date;
  endDate: Date;

  constructor(id: number, title: string, description: string, paintingIds: number[], startDate: Date, endDate: Date) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.paintingIds = paintingIds != null ? paintingIds : [];
    this.startDate = startDate;
    this.endDate = endDate;
  }
}

File: IagoAuction/DTOs/AuctionDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Threading.Tasks;

namespace IagoAuction.DTOs
{
    public class AuctionDto
    {
        public int Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public List<int> PaintingIds { get; set; }

    }
}
